cdap-security/src/main/java/io/cdap/cdap/security/authorization/AuthorizerClassLoader.java

```diff
@@ -72,7 +78,24 @@ public boolean acceptPackage(String packageName) {
     }
   }
 
-  AuthorizerClassLoader(File unpackedJarDir, @Nullable String authorizerExtraClasspath) {
-    super(unpackedJarDir, authorizerExtraClasspath, createParent(), "lib");
+  AuthorizerClassLoader(File tmpDir, File authorizerExtensionJar,
+                        @Nullable String authorizerExtraClasspath) throws IOException {
+    super(BundleJarUtil.prepareClassLoaderFolder(Locations.toLocation(authorizerExtensionJar), tmpDir),
+          authorizerExtraClasspath, createParent(), "lib");
+    this.tmpDir = tmpDir;
+    this.extensionJar = authorizerExtensionJar;
+  }
+
+  @Override
+  public void close() throws IOException {
+    super.close();
+    DirUtils.deleteDirectoryContents(tmpDir);
```

Comment:
nit: do it in finally in case super fails for whatever reason

Reply:
sg.